glean-core/tests/dual_labeled_counter.rs

```diff
@@ -525,3 +483,46 @@ fn caching_metrics_with_dynamic_labels_across_pings() {
     assert_eq!(json!(20), cached_labels["key20"]["category20"]);
     assert_eq!(json!(null), cached_labels["__other__"]["__other__"]);
 }
+
+#[test]
+fn labels_containing_a_record_separator_record_an_error() {
+    let (glean, _t) = new_glean(None);
+    let dual_labeled_counter = DualLabeledCounterMetric::new(
+        CommonMetricData {
+            name: "dual_labeled_counter".into(),
+            category: "telemetry".into(),
+            send_in_pings: vec!["store1".into()],
+            disabled: false,
+            lifetime: Lifetime::Ping,
+            ..Default::default()
+        },
+        None,
+        None,
+    );
+
+    let metric = dual_labeled_counter.get("record\x1eseparator", "category1");
+    metric.add_sync(&glean, 1);
+
+    let snapshot = StorageManager
+        .snapshot_as_json(glean.storage(), "store1", true)
+        .unwrap();
+
+    // We end up with 1x InvalidLabel error because the key contains an ASCII record separator, but both
+    // labels end up as other because the use of the record separator prevents us from knowing which one
+    // contained it (or both) at this point in exection.
```

Comment:
damn. Guess that's truly an edge case, so not worth "fixing". In the future we might properly store labels separately and then we don't need to care about separators anymore.